pstoreds/addr_book.go

```diff
@@ -309,57 +452,72 @@ func newTTLManager(parent context.Context, d ds.Datastore, c *lru.ARCCache, tick
 }
 
 // To be called by TTL manager's coroutine only.
-func (mgr *ttlmanager) tick() {
+func (mgr *ttlManager) tick() {
 	mgr.Lock()
 	defer mgr.Unlock()
 
 	now := time.Now()
-	batch, err := mgr.ds.Batch()
-	if err != nil {
-		log.Error(err)
+	var toDel []ds.Key
+	for key, entry := range mgr.entries {
+		if entry.ExpiresAt.After(now) {
+			continue
+		}
+		toDel = append(toDel, key)
+	}
+
+	if len(toDel) == 0 {
 		return
 	}
-	for key, entry := range mgr.entries {
-		if entry.ExpiresAt.Before(now) {
-			if err := batch.Delete(key); err != nil {
-				log.Error(err)
-			} else {
-				mgr.cache.Remove(key)
-			}
-			delete(mgr.entries, key)
+
+	txn := mgr.ds.NewTransaction(false)
+	defer txn.Discard()
+
+	for _, key := range toDel {
+		if err := txn.Delete(key); err != nil {
+			log.Error("failed to delete TTL key: %v, cause: %v", key.String(), err)
+			break
 		}
+		mgr.cache.Remove(key.Parent().Name())
+		delete(mgr.entries, key)
 	}
-	err = batch.Commit()
-	if err != nil {
-		log.Error(err)
+
+	if err := txn.Commit(); err != nil {
+		log.Error("failed to commit TTL deletion, cause: %v", err)
+	}
+}
+
+func (mgr *ttlManager) deleteTTLs(keys []ds.Key) {
+	mgr.Lock()
+	defer mgr.Unlock()
+
+	for _, key := range keys {
+		delete(mgr.entries, key)
 	}
 }
 
-func (mgr *ttlmanager) setTTLs(keys []ds.Key, ttl time.Duration, add bool) {
+func (mgr *ttlManager) insertTTLs(keys []ds.Key, ttl time.Duration) {
```

Comment:
i think insert/set should have their names switched?

Reply:
Naming is hard 😂  In hindsight I'd keep setTTLs as-is (it unconditionally overrides the ttl), and rename `insertTTLs` to `insertOrExtendTTLs`, to illustrate that it either inserts the TTL if it doesn't exist, or extends the TTL if it does; but it won't bring the expiration closer if it already exists. How does that sound?